storage/integration_test.go

```diff
@@ -5366,23 +5364,32 @@ func TestIntegration_BucketIPFilter(t *testing.T) {
 			t.Errorf("got bucket IPFilter %+v, want %+v", attrs.IPFilter, want)
 		}
 
-		// Now clear VPCNetworkSource and PublicNetworkSource.
+		// Clear VPCNetworkSource and PublicNetworkSource.
 		want = &IPFilter{
 			Mode: "Disabled",
-			// VPCNetworkSource and PublicNetworkSource omitted/zeroed to clear them.
 		}
 		ua = BucketAttrsToUpdate{
 			IPFilter: want,
 		}
 		attrs = h.mustUpdateBucket(bucket, ua, attrs.MetaGeneration)
 		if attrs.IPFilter == nil {
 			t.Errorf("got nil IPFilter, want non-nil")
-		}
-		if len(attrs.IPFilter.VPCNetworkSource) != 0 {
-			t.Errorf("expected VPCNetworkSource to be cleared, got %+v", attrs.IPFilter.VPCNetworkSource)
-		}
-		if attrs.IPFilter.PublicNetworkSource != nil && len(attrs.IPFilter.PublicNetworkSource.AllowedIPCidrRanges) != 0 {
-			t.Errorf("expected PublicNetworkSource to be cleared, got %+v", attrs.IPFilter.PublicNetworkSource)
+		} else {
+			// Retry in case of metadata propagation delay.
+			err := retry(ctx, func() error {
+				return nil
+			}, func() error {
+				if len(attrs.IPFilter.VPCNetworkSource) != 0 {
+					return fmt.Errorf("expected VPCNetworkSource to be cleared, got %+v", attrs.IPFilter.VPCNetworkSource)
+				}
+				if attrs.IPFilter.PublicNetworkSource != nil && len(attrs.IPFilter.PublicNetworkSource.AllowedIPCidrRanges) != 0 {
+					return fmt.Errorf("expected PublicNetworkSource to be cleared, got %+v", attrs.IPFilter.PublicNetworkSource)
+				}
+				return nil
+			})
+			if err != nil {
+				t.Error(err)
```

Comment:
Instead of calling t.Error from here I think we need to return the error and check the error returned from `retry`.